### PR TITLE
Wagtail 6.0

### PR DESCRIPTION
**Pull request by katdom13**
[Support Ticket](https://torchbox.monday.com/boards/1124794299/pulses/1396935446)

### Summary of changes

- Added Wagtail 6.0 compatibility
- Added tests for Django 5.0
- Added tests for Python 3.12

### CODE REVIEW

File: src/wagtailmarkdown/widgets.py [hunk 1]
```diff
@@ -6,37 +6,75 @@
 from wagtail.widget_adapters import WidgetAdapter
```
[the remark this answers is not in the document]
Comment (katdom13) [reply src/wagtailmarkdown/widgets.py:5]: Hi @zerolab,
Shouldn't it only be the import `from wagtail.utils.widgets import WidgetWithScript`?

File: src/wagtailmarkdown/widgets.py [hunk 2]
```diff
@@ -6,37 +6,75 @@
...
-            hasattr(settings, "WAGTAILMARKDOWN")
-            and "autodownload_fontawesome" in settings.WAGTAILMARKDOWN
-        ):
-            autodownload = (
-                "true"
-                if settings.WAGTAILMARKDOWN["autodownload_fontawesome"]
-                else "false"
+if WAGTAIL_VERSION >= (6, 0):
+
+    class MarkdownTextarea(WidgetWithScript, forms.widgets.Textarea):  # type: ignore PGH003
```
[the remark this answers is not in the document]
Comment (katdom13) [reply src/wagtailmarkdown/widgets.py]: Thanks, I forgot to remove this

File: src/wagtailmarkdown/widgets.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zerolab) [reply src/wagtailmarkdown/widgets.py:5]: erm, yes. that's what I meant

[the remark this answers is not in the document]
Comment (katdom13) [reply src/wagtailmarkdown/widgets.py:5]: Sorry @zerolab ,
It shows up as `from wagtail.widget_adapters import WidgetAdapter` for me.
I made the changes here along with fixing anything that affects the tests:
https://github.com/torchbox/wagtail-markdown/pull/139/commits/44f1a55f928821ee7e4bb1c7d181f5dc88feef52

Thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+- Support for the Django 5.0
+- Support for the Wagtail 6.0
+- Dropped support for Wagtail < 5.2
```
Comment (zerolab) [inline CHANGELOG.md:7]: note: these should follow the format used in other release notes.

Leave them as is, as I will reword them anyway

**Review by zerolab — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (katdom13) [reply CHANGELOG.md:7]: Thanks

File: src/wagtailmarkdown/static/wagtailmarkdown/js/easymde-controller.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// wagtailmarkdown/static/wagtailmarkdown/js/easymde-controller.js
```
Comment (lb-) [inline src/wagtailmarkdown/static/wagtailmarkdown/js/easymde-controller.js:1]: @zerolab @katdom13 - I have noticed this a bit in some other 6.0 migrations.

There is no need to put a comment with the filename in the file itself. The documentation just has this to show you an example of where you file might be located.

**Pull request merged**